PRCS/prc-4.md

```diff
@@ -0,0 +1,429 @@
+    /// by providing an exact amount of ETH and requesting a specific minimum amount of asset to receive.
+    /// @dev Transfers portions of msg.value to the orders' sellers according to the price.
+    /// The sum of asset amounts of filled orders MUST be at least `minimumAsset`.
+    /// If msg.value is more than the sum of orders' prices, it MUST refund the excess back to `msg.sender`.
+    /// MUST decrease the `assetAmount` parameter for the specified order according to how much of it was filled,
+    /// and transfer that amount of the order's `asset` with ID `assetId` to the buyer.
+    /// MUST emit `OrderFilled` event for each order accordingly.
+    function fillOrdersExactEth(
+        address asset,
+        uint256 minimumAsset,
+        uint256[] memory orderIds
+    ) external payable;
+
+    /// @notice Consecutively fills an array of orders identified by the asset-specific `orderId` of each order,
+    /// by providing a possibly surplus amount of ETH and requesting an exact amount of asset to receive.
+    /// @dev Transfers portions of msg.value to the orders' sellers according to the price.
+    /// The sum of asset amounts of filled orders MUST be exactly `assetAmount`. Excess ETH MUST be returned back to `msg.sender`.
+    /// MUST decrease the `assetAmount` parameter for the specified order according to how much of it was filled,
+    /// and transfer that amount of the order's `asset` with ID `assetId` to the buyer.
+    /// MUST emit `OrderFilled` event for each order accordingly.
+    /// If msg.value is more than the sum of orders' prices, it MUST refund the difference back to `msg.sender`.
+    function fillOrdersExactAsset(
+        address asset,
+        uint256 assetAmount,
+        uint256[] memory orderIds
+    ) external payable;
+
+    /// @notice Cancels the sell order of `asset` with asset-specific `orderId`, transferring the order's assets back to the seller.
+    /// @dev MUST revert if the order's seller is not `msg.sender`.
+    /// MUST change the `assetAmount` parameter for the specified order to `0`.
+    /// MUST emit `OrderCancelled` event.
+    /// MUST transfer the order's `assetAmount` of `asset` with `assetId` back to the seller.
+    function cancelSellOrder(address asset, uint256 orderId) external;
+
+    /// @notice Cancels a batch of sell orders of `asset` with asset-specific `orderIds`, transferring the orders' assets back to the seller.
+    /// @dev This is a batched version of `cancelSellOrder` that simply iterates through the array to call said function.
+    function cancelBatchSellOrder(address asset, uint256[] memory orderIds) external;
+
+    /// @notice Withdraws the contract balance (containing collected fees) to the owner. Executable only by the owner.
+    /// @dev MUST transfer the entire contract balance to the owner.
+    /// MUST revert if called by unauthorized account.
+    /// MUST emit `FeesWithdrawn` event.
+    function withdrawFees() external;
+}
```

Comment:
Maybe we need to add the claim function and the rationale for why it's needed? (the rationale thread we had on Slack)

Reply:
Added!